cmake/CondaRecognizer.cmake

```diff
@@ -1,18 +1,9 @@
 # This cmake module determines if a conda environment is active.
 # If so, CMAKE_INSTALL_PREFIX is set to the environmental variable CONDA_PREFIX.
-# This automatic behavior can prevented by passing the option -DCMAKE_CONDA_IGNORE=TRUE
+# This automatic behavior can be overriden by manually specifying a different CMAKE_INSTALL_PREFIX.
 
-
-# Keep the previous value of CMAKE_INSTALL_PREFIX before a possible change to CONDA_PREFIX
-set(CMAKE_INSTALL_PREFIX_BEFORE_CONDA_PREFIX ${CMAKE_INSTALL_PREFIX})
-
-# Check if a conda environment is active
-if(DEFINED ENV{CONDA_PREFIX} AND NOT CMAKE_CONDA_IGNORE)
+# Check if user didn't override CMAKE_INSTALL_PREFIX and a conda environment is active
+if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT and DEFINED ENV{CONDA_PREFIX})
```

Comment:
Hi @tadeu, I tried using `CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT` before, and it didn't work. Your proposed changes here also does not work -- conda env ends up not recognized. I'll see what can be done in terms of using `CMAKE_INSTALL_PREFIX` to determine behavior.

Reply:
Just tried here this code:
~~~
if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
    message("CMAKE_INSTALL_PREFIX INITIALIZED TO DEFAULT")
endif()
~~~
without setting `CMAKE_INSTALL_PREFIX` in the command line and the message was not printed.

Reply:
OK, it looks like this variable `CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT` is set after `project` command. I was testing here with the `project` command after `include(CondaRecognizer)`.